test/e2e/storage/drivers/csi.go

```diff
@@ -0,0 +1,243 @@
+	// pods should be scheduled on the node
+	nodes := framework.GetReadySchedulableNodesOrDie(cs)
+	node := nodes.Items[rand.Intn(len(nodes.Items))]
+	h.driverInfo.Config.ClientNodeName = node.Name
+	h.driverInfo.Config.ServerNodeName = node.Name
+
+	config := h.driverInfo.Config
+	h.serviceAccount = csiServiceAccount(cs, config, "hostpath", false)
+	csiClusterRoleBindings(cs, config, false, h.serviceAccount, h.combinedClusterRoleNames)
+	role := csiControllerRole(cs, config, false)
+	csiControllerRoleBinding(cs, config, false, role, h.serviceAccount)
+	csiHostPathPod(cs, config, false, f, h.serviceAccount)
+}
+
+func (h *hostpathCSIDriver) CleanupDriver() {
+	By("uninstalling csi hostpath driver")
+	f := h.driverInfo.Framework
+	cs := f.ClientSet
+	config := h.driverInfo.Config
+
+	csiHostPathPod(cs, config, true, f, h.serviceAccount)
+	csiClusterRoleBindings(cs, config, true, h.serviceAccount, h.combinedClusterRoleNames)
+	role := csiControllerRole(cs, config, true)
+	csiControllerRoleBinding(cs, config, true, role, h.serviceAccount)
+	csiServiceAccount(cs, config, h.driverInfo.Name, true)
+}
+
+// gce-pd
+type gcePDCSIDriver struct {
+	controllerClusterRoles   []string
+	nodeClusterRoles         []string
+	controllerServiceAccount *v1.ServiceAccount
+	nodeServiceAccount       *v1.ServiceAccount
+
+	driverInfo DriverInfo
+}
+
+var _ TestDriver = &gcePDCSIDriver{}
+var _ DynamicPVTestDriver = &gcePDCSIDriver{}
+
+// InitGcePDCSIDriver returns gcePDCSIDriver that implements TestDriver interface
+func InitGcePDCSIDriver() TestDriver {
+	return &gcePDCSIDriver{
+		nodeClusterRoles: []string{
+			csiDriverRegistrarClusterRoleName,
+		},
+		controllerClusterRoles: []string{
+			csiExternalAttacherClusterRoleName,
+			csiExternalProvisionerClusterRoleName,
+		},
+		driverInfo: DriverInfo{
+			Name:        "csi-gce-pd",
+			FeatureTag:  "",
+			MaxFileSize: testpatterns.FileSizeMedium,
+			SupportedFsType: sets.NewString(
+				"", // Default fsType
+				"ext2",
+				"ext3",
+				"ext4",
+				"xfs",
+			),
+			IsPersistent:       true,
+			IsFsGroupSupported: true,
+			IsBlockSupported:   true,
+		},
+	}
+}
+
+func (g *gcePDCSIDriver) GetDriverInfo() *DriverInfo {
+	return &g.driverInfo
+}
+
+func (g *gcePDCSIDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
+	f := g.driverInfo.Framework
+	cs := f.ClientSet
+	config := g.driverInfo.Config
+	framework.SkipUnlessProviderIs("gce", "gke")
+	// TODO(#62561): Use credentials through external pod identity when that goes GA instead of downloading keys.
+	createGCESecrets(cs, config)
+	framework.SkipUnlessSecretExistsAfterWait(cs, "cloud-sa", config.Namespace, 3*time.Minute)
+}
+
+func (g *gcePDCSIDriver) GetDynamicProvisionStorageClass(fsType string) *storagev1.StorageClass {
+	f := g.driverInfo.Framework
+	cs := f.ClientSet
+	ns := g.driverInfo.Framework.Namespace.Name
+	provisioner := g.driverInfo.Name
+	suffix := fmt.Sprintf("%s-sc", g.driverInfo.Name)
+
+	// pods should be scheduled on the node
```

Comment:
not sure what this is for but I don't think its necessary for GCE PD. We can schedule test pods on any node

Reply:
@davidz627 can you help test this PR on the gce environment?

I agree we should not be setting nodes here